src/main/java/entities/DesignableMaze.java

```diff
@@ -12,4 +16,65 @@ public void placeWall(int row, int col) {
     public void deleteWall(int row, int col) {
         this.state[row][col] = Maze.ENCODING.get("empty");
     }
-}
+
+    public void placeStart(int row, int col) {
+        for (int i = 0; i < this.getNumRow(); i++) {
+            for (int j = 0; j < this.getNumCol(); j++) {
+                if (this.state[i][j] == 'S'){
+                    this.state[i][j] = Maze.ENCODING.get("empty");
+                }
+            }
+        }
+        this.state[row][col] = Maze.ENCODING.get("start");
+        startLocation[0] = row;
+        startLocation[1] = col;
+    }
+    public void placeEnd(int row, int col) {
+        for (int i = 0; i < this.getNumRow(); i++) {
+            for (int j = 0; j < this.getNumCol(); j++) {
+                if (this.state[i][j] == 'E'){
```

Comment:
Reference the encoding instead of writing 'E" directly

Reply:
Fixed along with line 23.